aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -188,6 +191,36 @@ module aptos_framework::code {
         publish_package(owner, util::from_bytes<PackageMetadata>(metadata_serialized), code)
     }
 
+    /// Allowing the owner of a package to upload source code after the package was published.
+    /// The source code will be mapped to existing list of modules in PackageRegistry by name. New modules and missing
+    /// ones will be ignored.
+    public entry fun add_source_code(
+        owner: &signer,
+        package_name: String,
+        modules: vector<vector<u8>>,
+    ) acquires PackageRegistry {
+        let package_registry = borrow_global_mut<PackageRegistry>(signer::address_of(owner));
+        let (found, i) = vector::find(&package_registry.packages, |p| {
+            let p: &PackageMetadata = p;
+            p.name == package_name
+        });
+        assert!(found, error::invalid_argument(EPACKAGE_NOT_FOUND));
+        let package_metadata = vector::borrow_mut(&mut package_registry.packages, i);
+        vector::for_each(modules, |serialized_module| {
+            let m = util::from_bytes<ModuleMetadata>(serialized_module);
+            let source = m.source;
+            let name = m.name;
+            let (found, index) = vector::find(&package_metadata.modules, |m| {
+                let m: &ModuleMetadata = m;
+                m.name == name
+            });
+            if (found) {
```

Comment:
Should this not be an assert? It's strange to just silently drop module updates that aren't found.

Reply:
Good point. I was thinking it's more symmetric as this allows missing modules as well. But assertion would make it more clear there are extra modules involved